File: asignaturas/sql/Asignaturas-IvanRonco/Profesores_Y_Modulos_BD-Java/src/main/java/config/ConfigApp.java
package config;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.inject.Inject;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Properties;


public class ConfigApp {
    private static ConfigApp instance = null;
    @Inject
    public ConfigApp() {
        try {
            loadProperties();
            initStorage();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
    public static ConfigApp getInstance() {
        if (instance == null) instance = new ConfigApp();
        return instance;
    }

    private final Logger logger = LoggerFactory.getLogger(ConfigApp.class);

    public String APP_DATA;
    public String APP_AUTHOR;
    public String APP_VERSION;

    public String APP_URL;
    public boolean APP_INIT_DATABASE;


    public void loadProperties() throws IOException {
        logger.debug("Cargamos todas las propiedades de fichero 'config.properties'");

        Properties properties = new Properties();
        properties.load(new FileInputStream(ClassLoader.getSystemResource("config.properties").getFile()));

        APP_DATA = properties.getProperty("APP_DATA", "data");
        APP_DATA = System.getProperty("user.dir")+ File.separator+APP_DATA;
        APP_AUTHOR = properties.getProperty("APP_AUTHOR", "IvanRoncoCebadera");
        APP_VERSION = properties.getProperty("APP_VERSION", "1.0.0");

        APP_URL = properties.getProperty("APP_URL", "jdbc:sqlite:database.db");
        APP_INIT_DATABASE = properties.getProperty("APP_INIT_DATABASE", "true").equals("true") ;
    }

    private void initStorage() throws IOException {
        logger.debug("Cargamos, en caso de que no este ya, la carpeta donde almacenaremos nuestros ficheros.");
        Files.createDirectories(Path.of(APP_DATA));
    }


}

File: burguer-pig/sql/BurgerPig-IvanRonco/HamburgesasEIngredientes_DB-Kotlin/src/main/kotlin/config/ConfigApp.kt
package config

import mu.KotlinLogging
import service.database.ConfigDatabase
import java.io.File
import java.io.FileInputStream
import java.nio.file.Files
import java.util.Properties
import kotlin.io.path.Path

class ConfigApp(
    val configDatabase: ConfigDatabase
) {
    private val logger = KotlinLogging.logger {  }

    private val properties = Properties()

    val APP_DATA by lazy {
        properties.getProperty("APP_DATA") ?: "data"
    }
    val APP_AUTHOR by lazy {
        properties.getProperty("APP_AUTHOR") ?: "IvanRoncoCebadera"
    }
    val APP_VERSION by lazy {
        properties.getProperty("APP_VERSION") ?: "1.0.0"
    }

    init {
        loadProperties()
        initStorage()
        configDatabase.initDatabase()
    }

    private fun initStorage() {
        logger.debug { "Creamos, en caso de que no exista, la carpeta donde guardaremos los ficheros" }
        Files.createDirectories(Path(APP_DATA))
    }

    private fun loadProperties() {
        logger.debug { "Se cargan todas las propiedades de 'config.properties'" }
        properties.load(FileInputStream(ClassLoader.getSystemResource("config.properties").file))
    }
}